sys/net/routing/rpl/rpl.c

```diff
@@ -278,34 +278,35 @@ void *rpl_process(void *arg)
 
 void _rpl_update_routing_table(void)
 {
-    rpl_dodag_t *my_dodag = rpl_get_my_dodag();
-    rpl_routing_entry_t *rt;
+    rpl_dodag_t *my_dodag;
+    rpl_dodag_t *end;
+    rpl_routing_entry_t *rt = rpl_get_routing_table();
 
-    if (my_dodag != NULL) {
-        rt = rpl_get_routing_table();
+    for (uint8_t i = 0; i < rpl_max_routing_entries; i++) {
+        if (rt[i].used) {
+            if (rt[i].lifetime <= 1) {
+                memset(&rt[i], 0, sizeof(rt[i]));
+            }
+            else {
+                rt[i].lifetime = rt[i].lifetime - RPL_LIFETIME_STEP;
+            }
+        }
+    }
 
-        for (uint8_t i = 0; i < rpl_max_routing_entries; i++) {
-            if (rt[i].used) {
-                if (rt[i].lifetime <= 1) {
-                    memset(&rt[i], 0, sizeof(rt[i]));
+    for (my_dodag = rpl_dodags, end = my_dodag + RPL_MAX_DODAGS; my_dodag < end; my_dodag++) {
+        if (my_dodag->used) {
```

Comment:
Just for my curiosity, is this check needed? 
When the node joins a  DODAG the value is [set](https://github.com/cgundogan/RIOT/blob/rpl_update_rt_table_for_all_dodags/sys/net/routing/rpl/rpl_dodag.c#L105) and never touched again (if I'm not missed something).
I can't find any occurrences where the `->used` entry of a DODAG is set to 0.